assets/js/googlesitekit/modules/create-submit-changes-store.js

```diff
@@ -0,0 +1,201 @@
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @return {Object} Empty object on success, object with `error` property on failure.
+		 */
+		*submitChanges() {
+			const { dispatch } = yield Data.commonActions.getRegistry();
+			dispatch( storeName ).clearError( 'submitChanges', [] );
+
+			yield {
+				type: START_SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			const result = yield {
+				type: SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			if ( result.error ) {
+				yield dispatch( storeName ).receiveError( result.error, 'submitChanges', [] );
+			}
+
+			yield {
+				type: FINISH_SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			return result;
+		},
+	};
+
+	const reducer = ( state, { type } ) => {
+		switch ( type ) {
+			case START_SUBMIT_CHANGES: {
+				return {
+					...state,
+					hasStartedSubmittingChanges: true,
+					hasFinishedSubmittingChanges: false,
+					isDoingSubmitChanges: true,
+				};
+			}
+
+			case FINISH_SUBMIT_CHANGES: {
+				return {
+					...state,
+					hasStartedSubmittingChanges: false,
+					hasFinishedSubmittingChanges: true,
+					isDoingSubmitChanges: false,
+				};
+			}
+
+			default: {
+				return state;
+			}
+		}
+	};
+
+	const controls = {
+		[ SUBMIT_CHANGES ]: submitChanges,
+	};
+
+	const resolvers = {};
+
+	const {
+		safeSelector: canSubmitChanges,
+		dangerousSelector: validateSubmitChanges,
+	} = createValidationSelector( validateCanSubmitChanges );
+
+	const selectors = {
+		canSubmitChanges,
+		validateSubmitChanges,
+		...createStateSelectors( Object.keys( initialState ) ),
+	};
+
+	return {
+		initialState,
+		actions,
+		controls,
+		reducer,
+		resolvers,
+		selectors,
+	};
+}
+
+/**
+ * Creates a default submitChanges control function.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} slug      Module slug.
+ * @param {string} storeName Datastore slug.
+ * @return {Function} Control function to submit changes.
+ */
+export function makeDefaultSubmitChanges( slug, storeName ) {
+	return createRegistryControl( ( { select, dispatch } ) => async () => {
+		if ( select( storeName ).haveSettingsChanged() ) {
+			const { error } = await dispatch( storeName ).saveSettings();
+			if ( error ) {
+				return { error };
+			}
+		}
+
+		await API.invalidateCache( 'modules', slug );
+
+		// TODO: Remove once legacy dataAPI is no longer used.
+		invalidateCacheGroup( TYPE_MODULES, slug );
+
+		return {};
+	} );
+}
+
+/**
+ * Creates a default canSubmitChanges function.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} storeName Datastore slug.
+ * @return {Function} A function to check if settings can be submitted.
+ */
+export function makeDefaultCanSubmitChanges( storeName ) {
+	return ( select ) => {
+		const strictSelect = createStrictSelect( select );
+		const {
+			haveSettingsChanged,
+			isDoingSubmitChanges,
+		} = strictSelect( storeName );
+
+		invariant( ! isDoingSubmitChanges(), INVARIANT_DOING_SUBMIT_CHANGES );
+		invariant( haveSettingsChanged(), INVARIANT_SETTINGS_NOT_CHANGED );
+	};
+}
```

Comment:
`makeDefaultSubmitChanges` and `makeDefaultCanSubmitChanges` are defaults only for module stores that have settings, so I think these functions would be more suitable to place within `create-module-store.js` instead.

Reply:
@felixarntz I have moved it to the `create-settings-store.js` since it is settings related functionality. Does it work for you?

Reply:
Sounds good!